R/module_teal_data.R

```diff
@@ -61,24 +61,45 @@ srv_teal_data <- function(id,
   moduleServer(id, function(input, output, session) {
     logger::log_debug("srv_teal_data initializing.")
 
-    data_out <- if (is_arg_used(data_module$server, "data")) {
-      data_module$server(id = "data", data = data)
-    } else {
-      data_module$server(id = "data")
-    }
+    validated_data <- reactive(tryCatch(data(), error = function(e) e))
+
+    final_data <- reactive({
+      data <- validated_data()
+      if (inherits(data, "shiny.silent.error")) {
+        return(teal_data())
+      }
+      data
+    })
+
+    data_out <- tryCatch(
```

Comment:
Why tryCatch here? If the `$server` fails then app shouldn't start I guess, this failure is an initialization bug not a reactivity issue. I think we don't need a tryCatch, am I right?

Reply:
Yes, we do not need this. Thanks.